lib/mix/test/mix/tasks/test_test.exs

```diff
@@ -183,6 +183,24 @@ defmodule Mix.Tasks.TestTest do
                """
       end)
     end
+
+    test "exits with signal 1 when coverage is below threshold" do
+      in_fixture("cover_threshold_below_test", fn ->
+        assert {_, 1} = mix_code(["test", "--cover"])
+      end)
+    end
+
+    test "exits with signal 0 when coverage is equal to or above threshold" do
+      in_fixture("umbrella_test", fn ->
+        assert {_, 0} = mix_code(["test", "--cover"])
+      end)
+    end
+
+    test "exits with signal 0 when summary is false" do
+      in_fixture("cover_threshold_off_test", fn ->
+        assert {_, 0} = mix_code(["test", "--cover"])
+      end)
+    end
```

Comment:
Those tests are very expensive, so we want to keep them to a minimum. We also generally prefer to change existing features than adding a new feature per case, so I propose:

1. For checking status of 0, please change the existing tests that already call --cover

2. For checking the status of 1, get one of the existing fixtures and change the test files so coverage is low (for example, you can remove one of the test files)

3. There is no need to test what happens when summary is false

At the end no new fixtures should be necessary!